controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,684 @@
+	if err != nil {
+		fmt.Errorf("Failed to update cluster: %v", err)
+	}
+
+	clusterState := *result.ManagedClusterProperties.ProvisioningState
+	if clusterState == "Succeeded" {
+		logrus.Infof("Cluster [%s] was updated successfully", config.Name)
+	}
+
+	logrus.Infof("Cluster config [%s] was checked", config.Spec.DisplayName)
+	return config, nil
+}
+
+func newClientAuthorizer(spec aksv1.AKSClusterConfigSpec) (autorest.Authorizer, error) {
+	authBaseURL := spec.AuthBaseURL
+	if authBaseURL == "" {
+		authBaseURL = azure.PublicCloud.ActiveDirectoryEndpoint
+	}
+
+	oauthConfig, err := adal.NewOAuthConfig(authBaseURL, spec.TenantID)
+	if err != nil {
+		return nil, err
+	}
+
+	baseURL := spec.BaseURL
+	if baseURL == "" {
+		baseURL = azure.PublicCloud.ResourceManagerEndpoint
+	}
+
+	spToken, err := adal.NewServicePrincipalToken(*oauthConfig, spec.ClientID, spec.ClientSecret, baseURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return autorest.NewBearerAuthorizer(spToken), nil
+}
+
+func newResourceGroupsClient(authorizer autorest.Authorizer, spec aksv1.AKSClusterConfigSpec) (*resources.GroupsClient, error) {
+	if authorizer == nil {
+		newAuthorizer, err := newClientAuthorizer(spec)
+		if err != nil {
+			return nil, err
+		}
+
+		authorizer = newAuthorizer
+	}
+
+	baseURL := spec.BaseURL
+	if baseURL == "" {
+		baseURL = azure.PublicCloud.ResourceManagerEndpoint
+	}
+
+	client := resources.NewGroupsClientWithBaseURI(baseURL, spec.SubscriptionID)
+	client.Authorizer = authorizer
+
+	return &client, nil
+}
+
+func createResourceGroup(ctx context.Context, groupsClient *resources.GroupsClient, spec aksv1.AKSClusterConfigSpec) (bool, error) {
+	future, err := groupsClient.CreateOrUpdate(
+		ctx,
+		spec.ResourceGroup,
+		resources.Group{
+			Name:	  to.StringPtr(spec.ResourceGroup),
+			Location: to.StringPtr(spec.ResourceLocation),
+		})
+
+	if err != nil {
+		return false, fmt.Errorf("[AKS] error creating resource group '%s': %v", spec.ResourceGroup, err)
+	}
+
+	return future.StatusCode == 201, nil
+}
+
+func checkResourceGroup(ctx context.Context, groupsClient *resources.GroupsClient, spec aksv1.AKSClusterConfigSpec) (bool, error) {
+	resp, err := groupsClient.CheckExistence(ctx, spec.ResourceGroup)
+	if err != nil {
+		return false, fmt.Errorf("[AKS] error getting resource group '%s': %v", spec.ResourceGroup, err)
+	}
+
+	return resp.StatusCode == 204, nil
+}
+
+func removeResourceGroup(ctx context.Context, groupsClient *resources.GroupsClient, spec aksv1.AKSClusterConfigSpec) error {
+	future, err := groupsClient.Delete(ctx, spec.ResourceGroup)
+	if err != nil {
+		return fmt.Errorf("[AKS] error removing resource group '%s': %v", spec.ResourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, groupsClient.Client); err != nil {
+		logrus.Errorf("[AKS] cannot get the AKS cluster create or update future response: %v", err)
+		return err
+	}
+
+	logrus.Infof("Cluster %v removed successfully", spec.ClusterName)
```

Comment:
I'm not an AKS expert but this log seems like it does not belong in this function. Can you explain?